src/libs/select.c

```diff
@@ -220,30 +226,141 @@ static int lua_register_selection(lua_State *L)
   dt_lib_module_t *self = lua_touserdata(L, lua_upvalueindex(1));
   dt_lua_module_entry_push(L,"lib",self->plugin_name);
   lua_getuservalue(L,-1);
-  const char* key = luaL_checkstring(L,1);
-  luaL_checktype(L,2,LUA_TFUNCTION);
+  const char* name;
+  const char* key;
+  char* tmp_name = NULL;
+  // location of function argument on the stack: 2 - old format, 3 - new format (w/name)
+  int func_pos = 2;
+
+  if (lua_type(L, 3) == LUA_TFUNCTION)
+  {
+    name = luaL_checkstring(L,1);
+    key = luaL_checkstring(L,2);
+    func_pos = 3;
+  }
+  else
+  {
+    key = luaL_checkstring(L,1);
+    tmp_name = malloc(sizeof(key));
+    // replace spaces in name with underscores
+    for(int i = 0; key[i] != '\0'; i++)
```

Comment:
please wrap content of for loop with { and }.

Reply:
removed